### PR TITLE
Adding support for in-source builds

### PR DESCRIPTION
**Pull request by arnarthor**
This adds a check for in-source builds. 

There is one caveat to this PR that I would like to discuss. Since this uses `require` to read the bsconfig to check if the build should be done in-source, it requires the bsconfig to be valid JSON, which is not required by bucklescript since it ships it's own JSON parser.

If there's some way to use that JSON parser instead that would be great, otherwise this would need to be a limitation of using bs-loader (which IMO is fine, but could be confusing).

### CODE REVIEW

**Comment by rrdelaney** [comment]
Don't worry about the Travis build failing, there's some permission problems on the CI 😄

File: index.js [hunk 1]
```diff
@@ -69,13 +72,32 @@ const getCompiledFileSync = (moduleDir, path) => {
   return transformSrc(moduleDir, res.toString())
 }
 
+const isInSourceBuild = moduleOptions => {
```
Comment (rrdelaney) [inline index.js]: Can we use a Webpack config option instead like we do with the `module` field? We can switch both the reading the `bsconfig.json` at a later time

**Review by rrdelaney — CHANGES_REQUESTED** [review]
Thanks so much 😄 

I think we can skip parsing the bsconfig for now and just use a Webpack setting for in-source, maybe a separate release for reading config from the bsconfig

**Comment by arnarthor** [comment]
Yeah that sounds good 👍. We should probably document it then to make sure people don't forget to change their bsconfig when using this.

I'll add something about it as well.

**Comment by arnarthor** [comment]
I've fixed what we discussed (except I forgot to add docs). For some reason I can't get the tests to run locally any more (not even before I made the change). 

Anyways, this should be equivalent to the bsconfig parsing thing I did before. 👍

**Review by rrdelaney — APPROVED** [review]

**Comment by rrdelaney** [comment]
I'm just gonna see if I can get the tests running locally, then merge this in!

**Pull request merged**